src/Extensions/OrderExtension.php

```diff
@@ -63,29 +76,53 @@ public function paymentCaptured()
     }
 
     /**
-     *
      * @throws FlowException
      */
     public function scheduleOrder()
     {
         if (!$this->owner->Scheduled) {
             $xml = $this->formatDataForFlow();
 
-            // Create scheduled order object
-            $scheduledOrder = ScheduledOrder::create([
-                'OrderID' => $this->owner->ID,
-                'Active'  => 1,
-                'Status'  => FlowStatus::PENDING,
-                'XmlData' => $xml
-            ]);
+            // If it is false, we don't want to continue scheduling
+            if ($xml === false) {
+                // Add a status update
+                $statusUpdateData = [
+                    'NotifyCustomer'  => 0,
+                    'CustomerVisible' => 0,
+                    'Message'         => 'Order does not contain any Flow products: cancelling scheduled order.',
+                    'Status'          => OrderStatus::COMPLETED
+                ];
 
-            try {
-                $scheduledOrder->write();
-            } catch (ValidationException $e) {
-                throw new FlowException($e->getMessage(), $e->getCode());
+                $update = OrderStatusUpdate::create($statusUpdateData);
```

Comment:
Please split constructor and setting data into a separate `->update($statusUpdateData);` method calls. this breaks write occasionally (it thinks the data is loaded from the DB when it's not).